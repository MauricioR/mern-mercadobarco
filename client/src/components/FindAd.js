import React, { Component } from 'react';
import { TabContent, TabPane, Nav, NavItem, NavLink, Card, Button, CardTitle, CardText, Row, Col } from 'reactstrap';
import classnames from 'classnames';

class FindAd extends Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      activeTab: '1'
    };
  }

  toggle(tab) {
    if(this.state.activeTab !== tab) {
      this.setState({
        activeTab: tab
      });
    }
  }
  render() {
    return (
      <div>
        <Nav tabs>
          <NavItem>
            <NavLink
              className={classnames({ active: this.state.activeTab === '1'})}
              onClick={() => { this.toggle('1'); }}
              >
              Lanchas
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink
              className={classnames({ active: this.state.activeTab === '2'})}
              onClick={() => { this.toggle('2'); }}
              >
              Veleiros
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink
              className={classnames({ active: this.state.activeTab === '3'})}
              onClick={() => { this.toggle('3'); }}
              >
              Jet Skis
            </NavLink>
          </NavItem>
        </Nav>
        <TabContent activeTab={this.state.activeTab}>
          <TabPane tabId="1">
            <Row>
              <Col sm="12">
                <Card body>
                  <CardTitle>Lanchas</CardTitle>
                  <CardText>Filter</CardText>
                  <Button>Search</Button>
                </Card>
              </Col>
            </Row>
          </TabPane>
        </TabContent>
      </div>
    );
  }
}

export default FindAd;
