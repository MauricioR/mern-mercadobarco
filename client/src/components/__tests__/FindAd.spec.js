import React from 'react';
import { shallow } from 'enzyme';

import { TabContent, TabPane, Nav, NavItem, NavLink, Card, Button, CardTitle, CardText, Row, Col } from 'reactstrap';
import classnames from 'classnames';

import FindAd from '../FindAd';

describe('Testing FindAd Component', () => {
  it('should render correctly', () => {
    const component = shallow(<FindAd />);
    expect(component).toMatchSnapshot();
  });
});
