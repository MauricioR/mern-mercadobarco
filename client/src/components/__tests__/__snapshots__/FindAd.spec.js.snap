// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FindAd Component should render correctly 1`] = `
<div>
  <Nav
    tabs={true}
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        className="active"
        onClick={[Function]}
        tag="a"
      >
        Lanchas
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        className=""
        onClick={[Function]}
        tag="a"
      >
        Veleiros
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        className=""
        onClick={[Function]}
        tag="a"
      >
        Jet Skis
      </NavLink>
    </NavItem>
  </Nav>
  <TabContent
    activeTab="1"
    tag="div"
  >
    <TabPane
      tabId="1"
      tag="div"
    >
      <Row
        tag="div"
      >
        <Col
          sm="12"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Card
            body={true}
            tag="div"
          >
            <CardTitle
              tag="h5"
            >
              Lanchas
            </CardTitle>
            <CardText
              tag="p"
            >
              Filter
            </CardText>
            <Button
              color="secondary"
              tag="button"
            >
              Search
            </Button>
          </Card>
        </Col>
      </Row>
    </TabPane>
  </TabContent>
</div>
`;
